File: gaia-sdk-javascript/src/graphql/request/input/DeleteSkillImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";

/**
 * The specification to delete a Skill instance
 */
export class DeleteSkillImpulse {

    private tenantId:Uuid;
    private reference:Uuid;

    constructor (tenantId:Uuid, reference:Uuid) {
        this.tenantId = tenantId;
        this.reference = reference;
    }
}

File: gaia-sdk-python/gaia_sdk/graphql/request/input/DeleteSkillImpulse.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField


class DeleteSkillImpulse():
    """
    The specification to delete a Skill instance
    """
    tenantId: str
    reference: str

    def __init__(self, tenantId: str, reference: str):
        self.tenantId = tenantId
        self.reference = reference

    def __eq__(self, other):
        return self.tenantId == other.tenantId and self.reference == other.reference

    def __repr__(self):
        return {'tenantId': self.tenantId, 'reference': self.reference}
